client/pages/search-page.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import parseRoute from '../lib/parse-route';
+
+class Search extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
```

Comment:
I don't think this state is necessary, but you should make sure that you are keeping track of the user's input value in state.